ecl/hqlcpp/hqlsource.cpp

```diff
@@ -3108,8 +3108,18 @@ class DiskReadBuilder : public DiskReadBuilderBase
     DiskReadBuilder(HqlCppTranslator & _translator, IHqlExpression *_tableExpr, IHqlExpression *_nameExpr)
         : DiskReadBuilderBase(_translator, _tableExpr, _nameExpr, (_tableExpr->queryChild(2)->getOperator() != no_pipe))
     {
+        bool usesPluggableFiletype = (_tableExpr->queryChild(2)->getOperator() == no_filetype);
+
+        // Force generic disk read if a pluggable file type is used; testing
+        // specifically for hthor is fragile and this should be updated
+        if (translator.targetHThor() && usesPluggableFiletype)
+        {
+            // genericDiskReads = true; // not sure if this is needed
+            genericDiskRead = true;
+        }
+
         extractCanMatch = (modeOp == no_thor) || (modeOp == no_flat) ||
-                          ((modeOp == no_csv) && genericDiskRead);
+                          (((modeOp == no_csv) || modeOp == no_filetype) && genericDiskRead);
```

Comment:
This is clearer
```
(modeOp == nofiletype) ||
((modeOp == no_csv) && genericDiskRead)
```
Because this code is ensuring something is only done for csv files if they are being read through the generic activity.

Reply:
Fixed.